product-accelerators/apim/components/org.wso2.healthcare.apim.backendauth/src/main/java/org/wso2/healthcare/apim/backendauth/BackendAuthenticator.java

```diff
@@ -56,28 +62,35 @@ public boolean mediate(MessageContext messageContext) {
             log.debug("Backend authenticator mediator is invoked.");
         }
 
-        BackendAuthConfig config = backendAuthConfig.get(configName);
-        String accessToken;
-        BackendAuthHandler currentAuthenticator;
+        if (masterConfig.getClientId() != null && masterConfig.getClientId().contains("$")) {
+            masterConfig.setClientId(Utils.resolveConfigValues(this.clientId, messageContext));
+        }
+        if (masterConfig.getPrivateKeyAlias() != null && masterConfig.getPrivateKeyAlias().contains("$")) {
+            masterConfig.setPrivateKeyAlias(Utils.resolveConfigValues(this.keyAlias, messageContext));
+        }
```

Comment:
All this can be go inside the Utils.resolveConfigValues function, which only needs to pas the messageContext?

Reply:
updated in [f430b18](https://github.com/wso2/healthcare-accelerator/pull/15/commits/f430b18bd0b702de98625b692346053811d915e8)